app/gui/view/graph-editor/src/component/visualization/registry.rs

```diff
@@ -70,7 +70,11 @@ impl Registry {
             type_map.get(tp).cloned().unwrap_or_default();
         if tp != &any_type {
             if let Some(vis_for_any) = type_map.get(&any_type) {
-                result.extend(vis_for_any.iter().cloned());
+                for vis in vis_for_any {
+                    if !result.contains(vis) {
+                        result.push(vis.clone_ref());
+                    }
+                }
```

Comment:
We may easily speed it up by keeping hash set of paths. AFAIK the path identifies the visualization.